identity/credentials.go

```diff
@@ -237,7 +237,14 @@ func CredentialsEqual(a, b map[CredentialsType]Credentials) bool {
 			return false
 		}
 
-		if !reflect.DeepEqual(expect.Identifiers, actual.Identifiers) {
+		expectIdentifiers, actualIdentifiers := make(map[string]struct{}, len(expect.Identifiers)), make(map[string]struct{}, len(actual.Identifiers))
+		for _, i := range expect.Identifiers {
+			expectIdentifiers[i] = struct{}{}
+		}
+		for _, i := range actual.Identifiers {
+			actualIdentifiers[i] = struct{}{}
+		}
+		if !reflect.DeepEqual(expectIdentifiers, actualIdentifiers) {
```

Comment:
Doesn't this just check whether both arrays have the same length at the end, since you're just filling it with empty structs? It doesn't compare the actual credentials (anymore?), right?

Reply:
Looks like it uses i not as the index but the value - so the identifier. Should be good then? I guess this prevents issues with ordering?

Reply:
It's `for key, value := range []string{"abcde"}` so yes

Reply:
My question is more geared towards whether we can trust that both `a` & `b` contain the same set of credentials, because that is not tested anymore!

A better test would be to order these credentials by some indicator and do a deepequal then.

Reply:
Not sure if I follow - we tested the identifiers before and are testing them now also? The credential itself (config for example) is tested in the code above :)

Reply:
>The credential itself (config for example) is tested in the code above :)

I see, I think I understand the code now, but it's a bit confusing... Thanks for the explanation.